File: SQL Server 2012/Projects/Visual Basic/Chapter 23/ApClrObjects/Stored Procedures/GetInvoiceReport.vb
Imports System
Imports System.Data
Imports System.Data.SqlClient
Imports System.Data.SqlTypes
Imports Microsoft.SqlServer.Server


Partial Public Class StoredProcedures
    <Microsoft.SqlServer.Server.SqlProcedure()> _
    Public Shared Sub  GetInvoiceReport ()
        ' Open the Connection object for the context
        Dim connection As New SqlConnection("Context connection=true")
        connection.Open()

        ' Create a string that defines the Select statement
        Dim sql As String =
            "SELECT VendorName, InvoiceNumber, InvoiceDate, InvoiceTotal " &
            "FROM Invoices JOIN Vendors " &
            "ON Invoices.VendorID = Vendors.VendorID " &
            "WHERE InvoiceTotal - CreditTotal - PaymentTotal > 0"

        ' Create the Command object
        Dim selectCommand As New SqlCommand(sql, connection)

        ' Use the SqlPipe object to return the data
        Dim pipe As SqlPipe = SqlContext.Pipe
        pipe.ExecuteAndSend(selectCommand)

        ' Close the Connection object
        connection.Close()
    End Sub
End Class

File: SQL Server 2012/Projects/C#/Chapter 21/ApClrObjects/ApClrObjects/GetInvoiceReport.cs
//------------------------------------------------------------------------------
// <copyright file="CSSqlStoredProcedure.cs" company="Microsoft">
//     Copyright (c) Microsoft Corporation.  All rights reserved.
// </copyright>
//------------------------------------------------------------------------------
using System;
using System.Data;
using System.Data.SqlClient;
using System.Data.SqlTypes;
using Microsoft.SqlServer.Server;

public partial class StoredProcedures
{
    [Microsoft.SqlServer.Server.SqlProcedure]
    public static void GetInvoiceReport ()
    {
        // open the Connection object for the context
        SqlConnection connection =
            new SqlConnection("Context connection=true");
        connection.Open();

        // create a string that defines the select statement
        string selectStatement =
            "SELECT VendorName, InvoiceNumber, InvoiceDate, InvoiceTotal " +
            "FROM Invoices JOIN Vendors " +
            "ON Invoices.VendorID = Vendors.VendorID " +
            "WHERE InvoiceTotal - CreditTotal - PaymentTotal > 0";

        // create the Command object
        SqlCommand selectCommand =
            new SqlCommand(selectStatement, connection);

        // use the SqlPipe object to return the data
        SqlPipe pipe = SqlContext.Pipe;
        pipe.ExecuteAndSend(selectCommand);

        // close the Connection object
        connection.Close();
    }
}
